### PR TITLE
Another design for WAR-1133

### PR DESCRIPTION
**Pull request by brettleehari**
Developed two designs for the same tickets. 
The first one is available in : https://github.com/warriorframework/warriorframework/pull/142

### CODE REVIEW

File: warhorn/source/utils.py [hunk 1]
```diff
@@ -617,6 +602,22 @@ def install_depen(dependency, dependency_name, logfile, print_log_name,
...
+    filepath : The filepath from where version_package is derived.
+    """
+    version_package = False
+    if not os.path.exists(filepath):
+        filepath = "../requirements.txt"
+    with open(filepath, 'r') as requirements:
+        for line in requirements:
+            if package in line:
+                version_package = line
+    return version_package, os.path.abspath(filepath)
```
Comment (navvrnd) [inline warhorn/source/utils.py]: Please use this function to get only the version. May be create another function to get the actual file to be used.

File: warhorn/default_config.xml [hunk 2]
```diff
@@ -3,8 +3,8 @@
 <!-- To read full documentation for this file, please refer to user_generated/config_sample.xml -->
 
 <data>
-    <warhorn name="Warhorn">
-        <dependency name="jira" install="no" user="no"/>
+    <warhorn name="Warhorn" dependency_filepath="Provide filepath, where version details for packages below are defined. By default the versions defined in requirements.txt at the base directory would be used">
```
Comment (bjrr) [inline warhorn/default_config.xml]: Hi @brettleehari,

The design for the dependency_filepath attribute was not mentioned anywhere in the design document.

File: warhorn/default_config.xml [hunk 3]
```diff
@@ -3,8 +3,8 @@
...
+        <dependency name="jira" install="yes" user=""/>
```
Comment (bjrr) [inline warhorn/default_config.xml]: Hi @brettleehari,

Why was the "jira" dependency set to yes? We do not need the jira package any more - migrated to requests package many versions ago.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+jira==1.0.3
```
Comment (bjrr) [inline requirements.txt]: Hi @brettleehari,

The design document specifically mentions that the requirements.txt file should reside under warriorframework/warhorn and not directly under warriorframework. Keeping it under warriorframework directory implies that all the packages in the requirements.txt file are necessary for warrior to run - when that is not the case.

**Comment by bjrr** [comment]
This code does not follow the design spelled out in the design document and email attached to the JIRA ticket by:

1. Lets user install "extra" dependencies
2. Additional "dependency_filepath" attribute
3. requirements.txt not inside the correct directory.

**Comment by bjrr** [comment]
Hi @brettleehari,

The console logs for this feature were not attached to the Jira ticket.

**Comment by bjrr** [comment]
Hi @brettleehari,

The design document also mentions creating testcases to test this feature. These testcases have not been added to the pull request.

File: warhorn/default_config.xml (hunk 3 above)
Comment (brettleehari) [reply warhorn/default_config.xml]: It was a change while testing the ticket. Shall change

File: requirements.txt (hunk 4 above)
Comment (brettleehari) [reply requirements.txt]: @bjrr I am afraid that the design document clearly mentions what i have implemented. I do not agree with this comment.

**Comment by brettleehari** [comment]
@bjrr Kindly open the pull request from closed state to open up for discussion.

File: warhorn/default_config.xml (hunk 2 above)
Comment (brettleehari) [reply warhorn/default_config.xml]: I shall explain this over the call. We had discussed about this item.

File: requirements.txt (hunk 4 above)
Comment (brettleehari) [reply requirements.txt]: @bjrr Moreover, as per python pip requirement.txt conventions. The file is generally placed in the base directory. One possible change that we could do is to place it in warrior/ directory as these dependency apply for Warrior directory. Users who do not use warhorn assembler would only use requirements.txt having it in warhorn is not useful from user perspective. 

Main use of requirement.txt is pip funtionality of pip -r requirement.txt. I have designed it for warhorn and pip install -r <filename> (pythonic) usage. Both cases would work well :)

what do you think?

**Pull request closed without merging**